content/zh/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,264 @@
+
+在之前的 Ambient 网格实现中，应用程序 Pod 被添加到 Ambient 网格的方式如下：
+- `istio-cni` 节点代理检测现有或新启动的 Kubernetes Pod，
+  其命名空间被标记为 `istio.io/dataplane-mode=ambient`，表明它应包含在 Ambient 网格中。
+- 然后，`istio-cni` 节点代理在主机网络命名空间中建立网络重定向规则，
+  以便拦截进入或离开应用程序 Pod 的数据包，
+  并将其重定向到相关代理[端口](https://github.com/istio/ztunnel/blob/master/ARCHITECTURE.md#ports)（15008、15006 或 15001）。
+
+这意味着对于 Ambient 网格中 Pod 创建的数据包，该数据包将离开该源 Pod，
+进入节点的主机网络命名空间，然后理想情况下会被拦截并重定向到该节点的 ztunnel（在其自己的网络命名空间中运行）
+用于代理到目标 Pod，返回流程类似。
+
+这个模式作为初始 Ambient 网格 Alpha 版实现的简易设计工作得很好，
+但正如前面提到的，它有一个基本问题 - 具有许多 CNI 实现，并且在 Linux 中，
+有许多根本不同且不兼容的方法，您可以使用在其中配置数据包的方式从一个网络名称空间到另一个网络名称空间。
+您可以使用隧道、全覆盖网络、通过主机网络命名空间或绕过它。
+您可以浏览 Linux 用户空间网络堆栈，也可以跳过它并在内核空间堆栈中来回传输数据包，等等。
+对于每种可能的方法，也许都有一个利用它的 CNI 实现。
+
+这意味着使用之前的重定向方法，有很多 CNI 实现根本无法与 Ambient 配合使用。
+鉴于其对主机网络命名空间数据包重定向的依赖 - 任何不通过主机网络命名空间路由数据包的 CNI 都需要不同的重定向实现。
+即使对于明确这样实现的 CNI，我们也会遇到不可避免且可能无法解决的主机级规则冲突问题。
+我们是在 CNI 之前拦截，还是之后拦截？如果我们执行其中一项或另一项，
+一些 CNI 是否会崩溃，而他们却没有预料到这一点？由于 NetworkPolicy 必须在主机网络命名空间中强制执行，
+因此 NetworkPolicy 何时何地被强制执行？我们是否需要大量代码来对每个流行的 CNI 进行特殊处理？
+
+### Istio Ambient 流量重定向：新模式 {#istio-ambient-traffic-redirection-the-new-model}
+
+在新的 Ambient 模式中，应用程序 Pod 被添加到 Ambient 网格的方式如下：
+- `istio-cni` 节点代理检测到一个 Kubernetes Pod（现有的或新启动的），
+  其命名空间被标记为 `istio.io/dataplane-mode=ambient`，表明它应该包含在 Ambient 网格中。
+  - 如果启动了一个应添加到 Ambient 网格中的**新** Pod，
+    则 CRI 会触发 CNI 插件（由 `istio-cni` 代理安装和管理）。
+    该插件用于将新的 Pod 事件推送到节点的 `istio-cni` 代理，
+    并阻止 Pod 启动，直到代理成功配置重定向。由于 CNI 插件在 Kubernetes Pod 创建过程中尽可能早的被 CRI 调用，
+    这确保了我们可以尽早建立流量重定向，以防止启动期间流量逃逸，而无需依赖初始化容器之类的机制。
+  - 如果**已经运行**的 Pod 被添加到 Ambient 网格中，则会触发新的 Pod 事件。
+    `istio-cni` 节点代理的 Kubernetes API 观察程序会检测到这一点，并以相同的方式配置重定向。
+- `istio-cni` 节点代理进入 Pod 的网络命名空间，
+  并在 Pod 网络命名空间内建立网络重定向规则，以便拦截进入和离开 Pod 的数据包，
+  并将其透明地重定向到侦听[已知端口](https://github.com/istio/ztunnel/blob/master/ARCHITECTURE.md#ports)（15008、15006、15001）的节点本地 ztunnel 代理实例。
+- 然后，`istio-cni` 节点代理通过 Unix 域套接字通知节点 ztunnel，
+  它应该在 Pod 的网络命名空间内建立本地代理侦听端口
+  （在 15008、15006 和 15001 上），并为 ztunnel 提供低等级 Linux
+  [文件描述符](https://zh.wikipedia.org/wiki/File_descriptor)用来表示 Pod 的网络命名空间。
+  - 虽然套接字通常是由实际在该网络命名空间内运行的进程在 Linux 网络命名空间内创建的，
+    但完全可以利用 Linux 的低等级套接字 API 来允许在一个网络命名空间中运行的进程在另一个网络命名空间中创建侦听套接字，
+    假设目标网络命名空间在创建时是已知的。
+- 节点本地 ztunnel 在内部启动一个新的代理实例和侦听端口集，专用于新添加的 Pod。
+- 一旦 in-Pod 重定向规则就位并且 ztunnel 建立完成侦听端口，
+  Pod 就会被添加到网格中，并且流量开始像以前一样流经节点本地 ztunnel。
+
+下面是显示应用程序 Pod 添加到 Ambient 网格的基本流程图：
+
+{{< image width="100%"
+    link="./pod-added-to-ambient.svg"
+    alt="Pod 被添加到 Ambient 网格的流程"
+    >}}
+
+一旦 Pod 成功被添加到 Ambient 网格中，默认情况下，
+进出网格中 Pod 的流量将像 Istio 一贯的做法一样使用 mTLS 完全加密。
+
+现在，流量将作为加密流量进入和离开 Pod 网络命名空间 - 即使 Pod 中运行的用户应用程序对此一无所知，
+看上去 Ambient 网格中的每个 Pod 都能够执行网格策略并安全地加密流量。
+
+下图说明了新模式中 Ambient 网格中的 Pod 之间的加密流量如何流动：
+
+{{< image width="100%"
+    link="./traffic-flows-between-pods-in-ambient.svg"
+    alt="HBONE 流量在 Ambient 网格中 Pod 之间的流程"
+    >}}
+
+而且，和以前一样，对于有必要的用例，仍然可以处理来自网格外部的未加密的明文流量并强制执行策略：
+
+{{< image width="100%"
+    link="./traffic-flows-plaintext.svg"
```

Comment:
need a plan to localize these images

Reply:
Yes, plan to do it a little later.

Reply:
Great, that can be included in a separate PR.